src/BibleWell.App/Views/Components/TiptapRendererView.axaml.cs

```diff
@@ -1,11 +1,602 @@
+        {
+            foreach (var child in node.Content)
+            {
+                container.Children.Add(RenderNode(child));
+            }
+        }
+
+        return container;
+    }
+
+    private Inline RenderFootnoteInline(TiptapNode node)
+    {
+        var contentText = string.Join("", node.Content?.Select(RenderTextNodeTextOnly) ?? []);
+        if (string.IsNullOrWhiteSpace(contentText))
+        {
+            return new Run
+            {
+                Text = "",
+                Name = "TiptapFootnoteNoText"
+            };
+        }
+
+        var footnoteTextBlock = new TextBlock
+        {
+            Text = "*",
+            Classes =
+            {
+                "tiptap-footnote-text-block"
+            },
+            Name = "TiptapFootnoteText"
+        };
+
+        ToolTip.SetTip(footnoteTextBlock, contentText);
+        return new InlineUIContainer
+        {
+            Child = footnoteTextBlock,
+            Name = "TiptapFootnoteTextContainer"
+        };
+    }
+
+    private Control RenderHeading(TiptapNode node)
+    {
+        var flow = GetFlowDirection(node);
+        var headingText = string.Join("", node.Content?.Select(RenderTextNodeTextOnly) ?? []);
+
+        var fontSizeClass = node.Attrs?.Level switch
+        {
+            1 => "text-2xl",
+            2 => "text-xl",
+            3 => "text-lg",
+            _ => "text-base"
+        };
+
+        return new TextBlock
+        {
+            Text = headingText,
+            Margin = new Thickness(0, 2, 0, 1),
+            Classes =
+            {
+                fontSizeClass,
+                "h3"
+            },
+            FlowDirection = flow,
+            TextAlignment = GetTextAlignment(flow),
+            Name = "TiptapHeading"
+        };
+    }
+
+    private Control RenderParagraph(TiptapNode node)
+    {
+        var flow = GetFlowDirection(node);
+
+        var paragraph = new TextBlock
+        {
+            Classes =
+            {
+                "tiptap-paragraph"
+            },
+            FlowDirection = flow,
+            TextAlignment = GetTextAlignment(flow),
+            Inlines = [],
+            Margin = new Thickness(0, 4, 0, 4),
+            Name = "TiptapParagraph"
+        };
+
+        if (node.Content is not null)
+        {
+            foreach (var child in node.Content)
+            {
+                paragraph.Inlines.Add(RenderInlineText(child));
+            }
+        }
+
+        return paragraph;
+    }
+
+    private Inline RenderInlineText(TiptapNode node)
+    {
+        if (node.Type == "footnote")
+        {
+            return RenderFootnoteInline(node);
+        }
+
+        var text = node.Text ?? "";
+
+        // Look for resourceReference or bibleReference
+        if (node.Marks is not null)
+        {
+            foreach (var mark in node.Marks)
+            {
+                if (mark.Type == "resourceReference" && mark.Attrs is JsonElement rr)
```

Comment:
Same note as above about switching to a case-insensitive comparison.  Please also do an audit for other type string comparisons that are currently case-sensitive.